src/unix.rs

```diff
@@ -0,0 +1,97 @@
+use std::fs::File;
+use std::io;
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+
+use libc::{STDERR_FILENO, STDIN_FILENO, STDOUT_FILENO};
+
+pub(crate) use std::os::unix::io::{AsRawFd as AsRaw, IntoRawFd as IntoRaw, RawFd as Raw};
+
+pub(crate) fn as_raw(io: &impl AsRawFd) -> RawFd {
+    io.as_raw_fd()
+}
+pub(crate) fn into_raw(io: impl IntoRawFd) -> RawFd {
+    io.into_raw_fd()
+}
+
+pub(crate) fn override_stdin(io: RawFd, owned: bool) -> io::Result<File> {
+    override_stdio(STDIN_FILENO, io, owned)
+}
+pub(crate) fn override_stdout(io: RawFd, owned: bool) -> io::Result<File> {
+    override_stdio(STDOUT_FILENO, io, owned)
+}
+pub(crate) fn override_stderr(io: RawFd, owned: bool) -> io::Result<File> {
+    override_stdio(STDERR_FILENO, io, owned)
+}
+
+pub(crate) fn reset_stdin(old: RawFd) -> io::Result<()> {
+    set_stdio(0, old)
+}
+pub(crate) fn reset_stdout(old: RawFd) -> io::Result<()> {
+    set_stdio(1, old)
+}
+pub(crate) fn reset_stderr(old: RawFd) -> io::Result<()> {
+    set_stdio(2, old)
+}
```

Comment:
Why do you use literals here and not `STDIN_FILENO/STDOUT_FILENO/STDERR_FILENO` ?

Reply:
Ah, good point, I'll change it.